py/dml/toplevel.py

```diff
@@ -74,33 +74,41 @@ def determine_version(filestr, filename):
             raise ESYNTAX(SimpleSite(f"{filename}:{lineno}:{column}"),
                           None, "malformed DML version tag")
         version = (int(m.group('major')), int(m.group('minor')))
-        # Remove the language version tag, but preserve newlines.
-        # This helps us avoid having to parse the version tag in the
-        # grammar.
+        # Remove the language version tag, but preserve the correct
+        # number of characters. Note that some newlines may be
+        # converted to spaces in the string fed to the parser; this is
+        # OK since sites re-read the DML file to derive line numbers
+        # from file offsets
         ver_end = m.end(0)
         filestr = ' ' * ver_end + filestr[ver_end:]
     else:
-        # TODO: make this an error
-        report(WNOVER(SimpleSite(f"{filename}:1")))
-        version = (1, 2)
-        lineno = 1
-        column = 1
+        if compat.optional_version_statement in dml.globals.enabled_compat:
+            report(WNOVER(SimpleSite(f"{filename}:1")))
+            version = (1, 2)
+            lineno = 1
+            column = 1
+        else:
+            raise ESYNTAX(
+                SimpleSite(f"{filename}:1"), None,
```

Comment:
Consider breaking out the identical usages of `SimpleSite` to a variable defined before the `if`.

Reply:
No need to, duplication in dying code is harmless